libs/checksum/src/test/java/org/codice/ddf/checksum/Sha256ChecksumProviderTest.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.checksum;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectOutputStream;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import org.codice.ddf.checksum.impl.Sha256ChecksumProvider;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class Sha256ChecksumProviderTest {
+
+  private ChecksumProvider checksumProvider;
+
+  @Before
+  public void intialize() {
+    checksumProvider = new Sha256ChecksumProvider();
+  }
+
+  @Test
+  public void testCalculateChecksumString() throws IOException, NoSuchAlgorithmException {
+    String testString = "Hello World";
+    String checksumCompareHash = "cade3dbf41da60f7154ef628446ff163700b6a9860cd951de8e4321b2189de6d";
```

Comment:
( ❓ ) Can we validate these using a separate tool just to be sure? For example, I'm not getting consistent results leveraging Python when I expect I should. 
```
$ python -c "import hashlib; m = hashlib.sha256(); m.update('Hello World'); print(m.hexdigest());"
a591a6d40bf420404a011733cfb7b190d62c65bf0bcda32b57b277d9ad9f146e
```

Reply:
the tests are using an object stream for simplicity, so it's not a 1:1 encoding of the string 'Hello World'

Reply:
and the consumer uses it similarly https://github.com/codice/ddf/blob/master/catalog/plugin/catalog-plugin-checksum/src/main/java/org/codice/ddf/catalog/content/plugin/checksum/Checksum.java#L72